project/src/main/java/com/google/sps/data/Places.java

```diff
@@ -38,5 +45,31 @@
     return ImmutableList.copyOf(mutablePlaces);
   }
 
+  /**
+   * Filters the given list of places according to the given parameters.
+   * @param places the list we want to filter.
+   * @param minRating The minimal rating we want to limit the list by. A number between 1 and 5.
+   * @param filterIfNoWebsite Specifies whether we should filter if there is no available website
+   *        for the restaurant.
+   * @param filterBranchesOfSamePlace Specifies whether we should remove different branches of the
+   *        same place from the results. Places are considered the same if they have the same name.
+   * @return the filtered list.
+   */
+  public static ImmutableList<Place> filter(ImmutableList<Place> places, int minRating, Boolean
+      filterIfNoWebsite, Boolean filterBranchesOfSamePlace) {
+    //TODO(M1): also filter by place's status when the attribute is added
+    ImmutableList<Place> result =
+        places.stream()
+            .sorted(Comparator.comparing(Place::rating).reversed())
+            .filter(place -> place.rating() >= minRating)
+      //TODO(M1): take in account whether the place has a google maps link, when attribute is added
+            .filter(filterIfNoWebsite
+                ? place -> !Strings.isNullOrEmpty(place.websiteUrl())
+                : place -> true)
+            .collect(collectingAndThen(toCollection(() -> new TreeSet<>(comparing(Place::name))),
```

Comment:
haha you are absolutely right. Now I remembered why I didn't use this method in the first place (I ran to it when I was working on this) - because I didn't find an elegant way to add it with the condition. do you maybe have an idea? (aside for breaking the stream and starting a new one inside an 'if' statement).

Reply:
Breaking the stream is fine IMO:)